.github/workflows/aod-iam.yml

```diff
@@ -0,0 +1,51 @@
+# Workflow that validates and handles AOD IAM request.
+name: 'aod-iam'
+
+on:
+  pull_request:
+    types: ['opened']
+    branches:
+      - 'main'
+  pull_request_review:
+    types: ['submitted']
+
+jobs:
+  # Check if it is an AOD request
+  is_iam:
+    uses: 'abcxyz/aod-tempate/.github/workflows/reusable-checker.yml@main' # Since it does not have any access to github context, directly ping the tag/ref here.
+    with:
+      path: 'iam.yaml'
+
+  # Add always fail job.
+  always-fail:
```

Comment:
Not sure I understand why this is needed.

Reply:
To prevent the PR from merging to main.

Reply:
Ah ok, forgot about this

Reply:
Should give a step a more meaningful name